crates/forge/src/runner.rs

```diff
@@ -412,21 +414,81 @@ impl<'a> ContractRunner<'a> {
 
     /// Runs a single unit test.
     ///
-    /// Calls the given functions and returns the `TestResult`.
+    /// Calls the given function (and before test functions, if any) and returns the `TestResult`.
     ///
-    /// State modifications are not committed to the evm database but discarded after the call,
-    /// similar to `eth_call`.
+    /// Configured before test functions are called in order and state modifications committed to
+    /// the EVM database (therefore the unit test call will be made on modified state).
+    /// State modifications of before test functions and unit test function call are discarded after
+    /// test ends, similar to `eth_call`.
     pub fn run_unit_test(
         &self,
         func: &Function,
         should_fail: bool,
         setup: TestSetup,
     ) -> TestResult {
         let address = setup.address;
-        let test_result = TestResult::new(setup);
+        let mut test_result = TestResult::new(setup);
+        let mut executor = self.executor.clone();
+
+        // Apply before test configured functions (if any).
+        for ITest::BeforeTestSelectors { test_selector, before_selectors } in
```

Comment:
would move this to another function, and also check beforehand whether this function exists in the ABI to avoid needlessly calling it

Reply:
fixed in https://github.com/foundry-rs/foundry/pull/8497/commits/e3dea3e1f021fd46f924ca650344aa8d80adbb79 also used the same function to apply before test txes to fuzz tests as well